src/Databases/DatabaseReplicated.cpp

```diff
@@ -606,37 +608,92 @@ void DatabaseReplicated::createReplicaNodesInZooKeeper(const zkutil::ZooKeeperPt
                         "already contains some data and it does not look like Replicated database path.", zookeeper_path);
 
     /// Write host name to replica_path, it will protect from multiple replicas with the same name
-    auto host_id = getHostID(getContext(), db_uuid, cluster_auth_info.cluster_secure_connection);
+    const auto host_id = getHostID(getContext(), db_uuid, cluster_auth_info.cluster_secure_connection);
+
+    const std::vector<String> check_paths = {
+        replica_path,
+        replica_path + "/replica_group",
+        replica_path + "/digest",
+
+        /// Needed to mark all the queries
+        /// in the range (max log ptr at replica ZooKeeper nodes creation, max log ptr after replica recovery] as successful.
+        /// Previously, this method was not idempotent and max_log_ptr_at_creation could be stored in memory.
+        /// we need to store max_log_ptr_at_creation in ZooKeeper to make this method idempotent during replica creation.
+        replica_path + "/max_log_ptr_at_creation",
```

Comment:
Looks like it will fix https://github.com/ClickHouse/ClickHouse/pull/56796#discussion_r1421946837

Reply:
Actually, this wasn't the case before https://github.com/ClickHouse/ClickHouse/pull/69102/commits/8544718d329ff772af2e06b19d949a676fb84af0